stablehlo/dialect/VhloTypes.td

```diff
@@ -228,6 +228,49 @@ def VHLO_UniformQuantizedV1 : VHLO_TypeDef<"UniformQuantizedV1", "quant_v1", "0.
   }];
   let assemblyFormat = "`<` $storageType `` `:` `` $expressedType `,` $scale `` `:` `` $zeroPoint `,` $storageTypeMin `` `:` `` $storageTypeMax `,` $flags `>`";
 }
+def VHLO_QuantizationScalesV1 : ArrayRefParameter<"::llvm::APFloat", "array of double scales"> {
+  let parser = [{
+    [&]() -> FailureOr<llvm::SmallVector<::llvm::APFloat>> {
```

Comment:
Is there a reason we use APFloat while upstream uses double? Did this change happen after we added support? Is there a benefit to one over the other?

Reply:
Per communication with Jacques Pienaar, `Quantization dialect was one of the first, so it predates generating types `. 

IMO having `APFLoat` for VHLO dialect here allows us to use the [APFloat::hash_value()](https://github.com/llvm/llvm-project/blob/5b780c8c6c558ec283a9eec485a4f172df0f9fe1/llvm/include/llvm/ADT/APFloat.h#L1348) which otherwise needs to be computed explicitly ([here](https://github.com/llvm/llvm-project/blob/5ce2f73b2e5e6664d74b49ee45f11505f8306577/mlir/lib/Dialect/Quant/IR/TypeDetail.h#L131), Thanks @GleasonK for pointing to that).